asm/osMapTLBRdb.s

```diff
@@ -9,26 +10,27 @@
 
 .balign 16
 
-glabel osMapTLBRdb
-/* 0086E0 80007AE0 40085000 */  mfc0  $t0, $10
-/* 0086E4 80007AE4 2409001F */  li    $t1, 31
-/* 0086E8 80007AE8 40890000 */  mtc0  $t1, $0
-/* 0086EC 80007AEC 40802800 */  mtc0  $zero, $5
-/* 0086F0 80007AF0 240A0017 */  li    $t2, 23
-/* 0086F4 80007AF4 3C09C000 */  lui   $t1, 0xc000
-/* 0086F8 80007AF8 40895000 */  mtc0  $t1, $10
-/* 0086FC 80007AFC 3C098000 */  lui   $t1, 0x8000
-/* 008700 80007B00 00095982 */  srl   $t3, $t1, 6
-/* 008704 80007B04 016A5825 */  or    $t3, $t3, $t2
-/* 008708 80007B08 408B1000 */  mtc0  $t3, $2
-/* 00870C 80007B0C 24090001 */  li    $t1, 1
-/* 008710 80007B10 40891800 */  mtc0  $t1, $3
-/* 008714 80007B14 00000000 */  nop   
-/* 008718 80007B18 42000002 */  tlbwi 
-/* 00871C 80007B1C 00000000 */  nop   
-/* 008720 80007B20 00000000 */  nop   
-/* 008724 80007B24 00000000 */  nop   
-/* 008728 80007B28 00000000 */  nop   
-/* 00872C 80007B2C 40885000 */  mtc0  $t0, $10
-/* 008730 80007B30 03E00008 */  jr    $ra
-/* 008734 80007B34 00000000 */   nop   
+LEAF(osMapTLBRdb)
+    mfc0    $t0, C0_ENTRYHI
+    li      $t1, NTLBENTRIES
+    mtc0    $t1, C0_INX
+    mtc0    $zero, C0_PAGEMASK
+    li      $t2, (TLBLO_UNCACHED | TLBLO_D | TLBLO_V | TLBLO_G)
+    li      $t1, K2BASE
+    mtc0    $t1, C0_ENTRYHI
+    li      $t1, K0BASE
```

Comment:
this is the PFN so part of a physical address, does it make sense to use K0BASE for this?

Reply:
I believe `K0BASE` is the most appropriate define here. It has to be something that results in `lui $t1, 0x8000`, it's not totally clear to me why it's written this way at all though, and what the consequences are if any.

Reply:
What I mean is it seems like the **most** appropriate define, not that it seems to make a lot of sense.
I'm not sure if it's a bug or not or exactly the behaviour, my guess is it will translate virtual addresses to 0x80000000 addresses and output them over the SysAD bus to the RCP, which will try to find a device at that physical address. Maybe this worked on n64s that had the additional components needed to use the libultra rdb, but I don't know.

I took another look through the headers, specifically `rdb.h` which we don't have in the repo currently, and it has some defines:
```
/* U64 side address */
#define	RDB_BASE_REG		0xc0000000
#define RDB_WRITE_INTR_REG	(RDB_BASE_REG + 0x8)
#define RDB_READ_INTR_REG	(RDB_BASE_REG + 0xc)
#define RDB_BASE_VIRTUAL_ADDR	0x80000000
```

I guess since this produces the tlb mapping for the rdb, the `K2BASE` and `K0BASE` should really actually be `RDB_BASE_REG` and `RDB_BASE_VIRTUAL_ADDR`.

Reply:
then since we don't have `RDB_BASE_REG` and `RDB_BASE_VIRTUAL_ADDR`, I think at least `RDB_BASE_VIRTUAL_ADDR` should just be raw hex, using a virtual address as a physical address just doesn't make sense